sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -453,4 +482,112 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       .getRawOne();
     return !!offeringExists;
   }
+
+  /**
+   * Check if the given offering has any submitted applications.
+   * @param offeringId
+   * @returns true if an offering has any assessment
+   */
+  async hasExistingApplication(offeringId: number): Promise<boolean> {
+    const queryResult = await this.connection
+      .getRepository(StudentAssessment)
+      .createQueryBuilder("assessment")
+      .select("1")
+      .where("assessment.offering.id = :offeringId", { offeringId })
+      .limit(1)
+      .getRawOne();
+    return !!queryResult;
+  }
+
+  /**
+   * Request a change for an offering.
+   * @param locationId
+   * @param programId
+   * @param offeringId
+   * @param userId
+   * @param educationProgramOffering
+   * @returns new offering created from existing offering with changes requested.
+   */
+  async requestChange(
+    locationId: number,
+    programId: number,
+    offeringId: number,
+    userId: number,
+    educationProgramOffering: SaveOfferingModel,
+  ): Promise<EducationProgramOffering> {
+    const currentOffering = await this.getApprovedOfferingToRequestChange(
+      locationId,
+      programId,
+      offeringId,
+    );
+
+    if (!currentOffering) {
+      throw new CustomNamedError(
+        "Either offering for given location and program not found or the offering not in appropriate status to be requested for change.",
+        OFFERING_NOT_VALID,
+      );
+    }
+
+    const requestedOffering = this.populateProgramOffering(
+      locationId,
+      programId,
+      educationProgramOffering,
+    );
+    const auditUser = { id: userId } as User;
+    requestedOffering.offeringStatus = OfferingStatus.AwaitingApproval;
+    requestedOffering.parentOffering =
+      currentOffering.parentOffering ??
+      ({ id: currentOffering.id } as EducationProgramOffering);
+    requestedOffering.creator = auditUser;
+    requestedOffering.createdAt = new Date();
+    let persistedOffering = new EducationProgramOffering();
```

Comment:
No need to have it declared at this level, it can be returned from inside the transaction.